homeassistant/components/binary_sensor/isy994.py

```diff
@@ -80,11 +73,12 @@ def setup_platform(hass, config: ConfigType,
                 device = ISYBinarySensorDevice(node)
                 devices.append(device)
 
-    for program in isy.PROGRAMS.get(DOMAIN, []):
+    for program in hass.data[ISY994_PROGRAMS][DOMAIN]:
         try:
-            status = program[isy.KEY_STATUS]
-        except (KeyError, AssertionError):
-            pass
+            status = program[KEY_STATUS]
+        except (AttributeError, KeyError, AssertionError):
```

Comment:
There doesn't seem to be an assertion check within this try...except.

Would it be possible to move all these try...except blocks for program out of the platforms and into the `_categorize_programs` function in the component? The aim would be that all programs in `ISY994_PROGRAMS` should be programs that should be added as entities. That would save a lot of near duplicated code. I see the block here in the binary sensor doesn't access the `KEY_ACTIONS` key on `program`. But that should be solvable in `_categorize_programs` by checking the domain.

Reply:
Yeah, I agree that's a much better way for this to work. I'll move it over.